### PR TITLE
CM-1828 only fetch decrypted key once per encrypted key

### PR DESCRIPTION
**Pull request by lukedemi**

### CODE REVIEW

File: lib/logstash/filters/cipher.rb [hunk 1]
```diff
@@ -157,11 +157,15 @@ def filter(event)
         encryptedKey = Base64.strict_decode64(dataHash["encryptedKey"])
         encryptionContext = eval(dataHash['encryptionContext'])
 
-        decryptedKey = @kms.decrypt(:ciphertext_blob =>  encryptedKey,
-                                   encryption_context: encryptionContext
-                                  )
+        unless @keys_hash.has_key? encryptedKey
+          @logger.warn("YES FETCH")
+          @decryptedKey = @kms.decrypt(:ciphertext_blob =>  encryptedKey,
```
Comment (ngunia) [inline lib/logstash/filters/cipher.rb]: Two different syntax formats for the same thing, change this to ciphertext_blob: encryptedKey

**Comment by ngunia** [comment]
LGTM besides my small comment

File: lib/logstash/filters/cipher.rb (hunk 1 above)
Comment (ngunia) [inline lib/logstash/filters/cipher.rb]: Also, I don't think decryptedKey needs an @, it doesn't look like it's used outside of this method

File: lib/logstash/filters/cipher.rb [hunk 2]
```diff
@@ -157,11 +157,15 @@ def filter(event)
         encryptedKey = Base64.strict_decode64(dataHash["encryptedKey"])
         encryptionContext = eval(dataHash['encryptionContext'])
 
-        decryptedKey = @kms.decrypt(:ciphertext_blob =>  encryptedKey,
-                                   encryption_context: encryptionContext
-                                  )
+        unless @keys_hash.has_key? encryptedKey
+          @logger.warn("YES FETCH")
```
Comment (bflad) [inline lib/logstash/filters/cipher.rb:161]: Minor improvement: it'd be really nice if this was `@logger.info("Fetching KMS decrypted key for #{encryptedKey}")` in the logs

**Pull request merged**